### PR TITLE
Adjust Chpl requests

### PR DESCRIPTION
**Pull request by karenzheung**
Adds small delay between requests and add user agent to CHPL requests
If communication with CHPL was not successful change errors to warning so it doesn't break the system

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
- [x] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: 
https://oncprojectracking.healthit.gov/support/browse/LANTERN-240
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [x] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [x] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [x] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge
is complete.

Create a new branch `<branch name>_gomodupdate`. Run `make update_mods` to update the `go.mod`
and `go.sum` files to point to master. Create a PR. Require one sanity check reviewer.

### CODE REVIEW

File: endpointmanager/pkg/chplquerier/vendorquerier_integration_test.go [hunk 1]
```diff
@@ -198,6 +202,9 @@ func Test_GetCHPLVendors(t *testing.T) {
 	// test context ended
 	// also checks what happens when an http request fails
 
+	hook := logtest.NewGlobal()
+	expectedErr := "Got error:\nmaking the GET request to the CHPL server failed: Get \"https://chpl.healthit.gov/rest/developers?api_key=tmp_api_key\": context canceled\n\nfrom URL: https://chpl.healthit.gov/rest/developers?api_key=tmp_api_key"
+
 	tc, err = basicVendorTestClient()
 	th.Assert(t, err == nil, err)
 	defer tc.Close()
```
Comment (emichaud998) [inline endpointmanager/pkg/chplquerier/vendorquerier_integration_test.go:210]: Maybe just to keep things consistent, move the lines with variables hook and expectedErr to under where the test client is made, since this is the ordering in the other files containing this test

File: endpointmanager/pkg/chplquerier/vendorquerier_integration_test.go [hunk 2]
```diff
@@ -208,13 +215,37 @@ func Test_GetCHPLVendors(t *testing.T) {
...
+	}
+	th.Assert(t, found, "expected an error to be logged")
+
+	// test http status != 200
+
+	tc = th.NewTestClientWith404()
+	defer tc.Close()
+
+	hook = logtest.NewGlobal()
+	expectedErr = "CHPL request responded with status: 404 Not Found"
```
Comment (emichaud998) [inline endpointmanager/pkg/chplquerier/vendorquerier_integration_test.go:234]: Maybe just to keep things consistent, move the lines with variables hook and expectedErr to above the lines where the test client is made, since this is the ordering in the other files containing these kinds of tests

File: endpointmanager/pkg/chplquerier/vendorquerier_test.go [hunk 3]
```diff
@@ -339,32 +351,52 @@ func Test_getVendorJSON(t *testing.T) {
...
+	th.Assert(t, found, "expected a context canceled error to be logged")
 
 	// test http status != 200
 
 	tc = th.NewTestClientWith404()
 	defer tc.Close()
 
+	hook = logtest.NewGlobal()
+	expectedErr = "CHPL request responded with status: 404 Not Found"
+
```
Comment (emichaud998) [inline endpointmanager/pkg/chplquerier/vendorquerier_test.go:385]: Same comment as before, maybe just to keep things consistent, move the lines with variables hook and expectedErr to above the lines where the test client is made, since this is the ordering in the other files containing these kinds of tests.
